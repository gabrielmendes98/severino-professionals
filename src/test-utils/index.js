/* eslint-disable react/prop-types */
import { render, configure, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Form, Formik } from 'formik';
import MainProvider from 'commons/providers/MainProvider';

configure({ testIdAttribute: 'id' });

const Wrapper = ({ children }) => (
  <>
    <MainProvider>{children}</MainProvider>
    <div id="modal"></div>
  </>
);
const RouteWrapper = ({ children }) => (
  <BrowserRouter>
    <Wrapper>{children}</Wrapper>
  </BrowserRouter>
);
const FormikWrapper = ({ children, initialValues, ...config }) => (
  <Formik initialValues={initialValues} {...config}>
    <Form>{children}</Form>
  </Formik>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: Wrapper, ...options });

const renderWithRouter = (ui, { route = '/', ...options } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: RouteWrapper, ...options });
};

const asFragment = ui => {
  const { asFragment } = customRender(ui);
  return asFragment(ui);
};

const toMatchSnapshot = (Component, { useRouter = false } = {}) => {
  const wrapper = asFragment(
    useRouter ? <BrowserRouter>{Component}</BrowserRouter> : Component,
  );
  expect(wrapper).toMatchSnapshot();
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export {
  customRender as render,
  screen,
  toMatchSnapshot,
  asFragment,
  renderWithRouter,
  FormikWrapper,
};
