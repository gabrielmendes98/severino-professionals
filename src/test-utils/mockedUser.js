const mockedUser = {
  id: '9c6cb22e-f284-4ca4-9a88-d64ece7b896c',
  createdAt: '2021-09-02T03:55:52.338Z',
  updatedAt: '2021-09-02T03:55:52.338Z',
  email: 'test@test.com',
  name: 'Testerson da Silva',
  description: null,
  avatarUrl: 'www.avatarurlfake.com',
  phone: '(11) 11111-1112',
  state: 'Minas Gerais',
  city: '√Åguas Formosas',
  hasWhatsapp: false,
};
export default mockedUser;
