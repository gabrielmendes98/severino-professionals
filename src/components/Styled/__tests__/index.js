import { toMatchSnapshot } from 'test-utils';
import { Grid } from '..';

it('Styled Grid should match snapshot', () => {
  toMatchSnapshot(<Grid margin={{ bottom: 2 }} padding={{ top: 2 }}></Grid>);
});
