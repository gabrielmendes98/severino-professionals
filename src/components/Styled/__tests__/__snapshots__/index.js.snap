// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Grid should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  margin-top: 0px;
  margin-left: 0px;
  margin-right: 0px;
  margin-bottom: 16px;
  padding-top: 16px;
  padding-left: 0px;
  padding-right: 0px;
  padding-bottom: 0px;
}

<div
    class="MuiGrid-root c0"
  />
  <div
    id="modal"
  />
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
